gobblin-utility/src/main/java/org/apache/gobblin/util/FileListUtils.java

```diff
@@ -69,12 +69,32 @@ public boolean accept(Path path) {
   }
 
   /**
+   * Given a path to copy, list all files rooted at the given path to copy
+   *
+   * @param fs the file system of the path
+   * @param path root path to copy
+   * @param fileFilter a filter only applied to root
+   */
+  public static List<FileStatus> listFilesToCopyAtPath(FileSystem fs, Path path, PathFilter fileFilter) throws IOException {
```

Comment:
Maybe call it `listFilesAndEmptyDirectoriesAtPath`? `FileListUtils` is used by more than just distcp.